# Appendix II. Databases & Servers

## 1\) Projects

1. ENCODE

   **Encyclopedia of DNA Elements**: a comprehensive parts list of functional elements in the human genome. [https://www.encodeproject.org/](https://www.encodeproject.org/)

2. 1000 Genomes

   most genetic variants with frequencies of at least 1% in the populations studied. [http://www.internationalgenome.org/](http://www.internationalgenome.org/)

3. TCGA

   **The Cancer Genome Atlas:** From the analysis of over 11,000 tumors from 33 of the most prevalent forms of cancer, the Pan-Cancer Atlas provides a uniquely comprehensive, in-depth, and interconnected understanding of how, where, and why tumors arise in humans. As a singular and unified point of reference, the Pan-Cancer Atlas is an essential resource for the development of new treatments in the pursuit of precision medicine. ​[https://portal.gdc.cancer.gov/](https://portal.gdc.cancer.gov/)​

   ​[http://www.cell.com/pb-assets/consortium/pancanceratlas/pancani3/index.html](http://www.cell.com/pb-assets/consortium/pancanceratlas/pancani3/index.html)​

4. CCLE \(cancer cell lines\)

   **Cancer Cell Line Encyclopedia**: [https://portals.broadinstitute.org/ccle](https://portals.broadinstitute.org/ccle)

## 2\) Databases

### 2a\) General

1. NCBI

   contribute to the NIH mission of ‘uncovering new knowledge’. [https://www.ncbi.nlm.nih.gov/](https://www.ncbi.nlm.nih.gov/)

2. UCSC

   genome browser for vertebrate. [https://genome.ucsc.edu/](https://genome.ucsc.edu/)

3. Ensemble

   genome annotation. [http://www.ensembl.org/index.html](http://www.ensembl.org/index.html)

### 2b\) Expression Data

1. GTEx

   gene expression in different tissues. [https://www.gtexportal.org/home/](https://www.gtexportal.org/home/)

2. Expression Atlas

   exploring gene expression results across species under different biological conditions. [https://www.ebi.ac.uk/gxa/home](https://www.ebi.ac.uk/gxa/home)

3. GEPIA

   gene expression in different TCGA tumor types. [http://gepia.cancer-pku.cn/index.html](http://gepia.cancer-pku.cn/index.html)

4. cBioPortal

   visualization, analysis and download of large-scale cancer genomics data sets. [http://www.cbioportal.org/index.do](http://www.cbioportal.org/index.do)

5. TCGA ncRNA

   ​[http://ibl.mdanderson.org/tanric/\_design/basic/index.html](http://ibl.mdanderson.org/tanric/_design/basic/index.html)​

## 3\) Servers

### 3a\) General

1. BLAST

   finds regions of similarity between biological sequences. [https://blast.ncbi.nlm.nih.gov/Blast.cgi](https://blast.ncbi.nlm.nih.gov/Blast.cgi)

### 3b\) Gene functional analysis

1. David

   gene functional classification tool. [https://david.ncifcrf.gov/](https://david.ncifcrf.gov/)

2. KAAS

   KEGG Automatic Annotation Server. [https://www.genome.jp/kegg/kaas/](https://www.genome.jp/kegg/kaas/)

3. agriGO

   gene ontology \(GO\) enrichment analyses of plant and agricultural species. [http://systemsbiology.cau.edu.cn/agriGOv2/](http://systemsbiology.cau.edu.cn/agriGOv2/)

### 3c\) RNA secondary structure

1. RNAfold

   predict secondary structures of single stranded RNA or DNA sequences. [http://rna.tbi.univie.ac.at/cgi-bin/RNAWebSuite/RNAfold.cgi](http://rna.tbi.univie.ac.at/cgi-bin/RNAWebSuite/RNAfold.cgi)

2. RNAstructure

   RNA secondary structure prediction and analysis. [http://rna.urmc.rochester.edu/RNAstructureWeb/](http://rna.urmc.rochester.edu/RNAstructureWeb/)

### 3d\) miRNA targets prediction

1. TargetScan

   search for predicted microRNA targets in mammals [http://www.targetscan.org/vert\_72/](http://www.targetscan.org/vert_72/)

2. psRNATarget

   plant small RNA target analysis [https://plantgrn.noble.org/psRNATarget/](https://plantgrn.noble.org/psRNATarget/)

### 3e\) motif analysis
**sequence**
1. MEME  
motif based sequence analysis tools [http://meme-suite.org/](http://meme-suite.org/)
2. HOMER  
Software for motif discovery and next-gen sequencing analysis [http://homer.ucsd.edu/homer/motif/](http://homer.ucsd.edu/homer/motif/)

**structure**
3. RNApromo   
Computational prediction of RNA structural motifs involved in post transcriptional regulatory processes [https://genie.weizmann.ac.il/pubs/rnamotifs08/](https://genie.weizmann.ac.il/pubs/rnamotifs08/)
4. GraphProt  
modeling binding preferences of RNA-binding proteins [http://www.bioinf.uni-freiburg.de/Software/GraphProt/](http://www.bioinf.uni-freiburg.de/Software/GraphProt/)
